### PR TITLE
Add user's auth token when authenticating Twitter (#253)

### PR DESCRIPTION
**Pull request by jturner-22**
This PR addresses issue #253, or at least one instance of it.

When initiating the Twitter authentication process, I've added the Node-RED user's access token as a query parameter. This had to be added to both the initial request and the callback URL. A query parameter named "access_token" is supported by the [passport-http-bearer](https://www.npmjs.com/package/passport-bearer-strategy) module used by Node-RED. This then allows the RED.auth.needsPermission() check to pass on both of the Twitter node's server endpoints.

The only other ways to get through the "passport-http-bearer" check is with the user's bearer token in the HTTP header or the body. I don't think either of these would work with the approach used by the Twitter node.

However, I'm uncomfortable that the Node-RED user's bearer token is being passed in the callback URL. That's going to Twitter and back. If that is indeed unacceptable, we can close this PR.

Maybe this needs a cookie-based approach or Twitter's [PIN-based](https://dev.twitter.com/oauth/pin-based) option, both of which would be a big change.

Maybe a compromise would be to remove the authentication check on the callback endpoint.

### CODE REVIEW

**Comment by knolleary** [comment]
Agreed with your comment that we shouldn't be passing the access token on the request.

I think removing the access control on that specific route may be the best way to proceed.

Am on vacation this week - if you want to propose an alternative pr then I'll take a look when I can.

**Comment by jturner-22** [comment]
I reworked the fix a bit, and like it more than the first. Now neither of the requests has the user's authorization token in a query parameter.

 1. __First Request:__ On the original request at '/twitter-credentials/:id/auth' that starts the authentication process, I removed the authentication token from the query parameter.

    Instead, I replaced the "Click here to authenticate with Twitter" anchor's href link with a $.ajax() call. The server now sends a 200 with the Twitter URL that needs to be opened, rather a 302 redirect. The client then opens a new window at that URL.

    Basically, I'm doing my own redirect into a new window, rather than deal with the underlying XMLHttpRequest trying to handle the redirect.

2. __Callback Request:__ As suggested, I removed the authorization check on the callback which is called by Twitter's authorization page.

    This still doesn't seem ideal, but it seems to me that anything else will require much larger changes and is well beyond this immediate fix.

**Comment by knolleary** [comment]
On review, I think it is sufficient to remove the `needsPermission` middleware from both auth routes. I don't immediately see a way that can be abused in a meaningful way.

Have pushed that fix and published a new version of the node to npm.
